tests/ApiTest.php

```diff
@@ -464,6 +468,29 @@ public function testUploadFromStringWithCustomFilename()
         $this->assertEquals($file->data['original_filename'], "test.txt");
     }
 
+    /**
+     * Test uploading from path
+     */
+    public function testAuthenticatedUrls()
+    {
+        try {
+            $file = $this->authenticatedApi->uploader->fromPath(dirname(__FILE__).'/test.jpg', 'image/jpeg', 'rename.jpg');
+        } catch (Exception $e) {
+            $this->fail('We get an unexpected exception trying to upload from path: '.$e->getMessage());
+        }
+
+        $dargs = array("ssl" => array("verify_peer" => false, "verify_peer_name" => false), "http" => array('timeout' => 60, 'user_agent' => 'Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.1.9) Gecko/20071025 Firefox/3.0.0.1'));
+
+        try {
+            $this->assertEquals(
+                file_get_contents($file->getUrl(), false, stream_context_create($dargs)),
+                file_get_contents(dirname(__FILE__).'/test.jpg')
+            );
+        } catch (Exception $e) {
+            $this->fail('Error while compare file downloaded by authenticated url and the original : '.$e->getMessage());
```

Comment:
you will catch a lot of other things here, and the error message will be confusing and mask real reasons (e.g. for network errors).